### PR TITLE
fix 'Request hook added multiple times when using filter rule ' (close #2650)

### LINKED ISSUE #2650: Request hook added multiple times when using filter rule
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
If I initialize hook without filter like `const hook = new RequestLogger();` then `onRequest` and `onResponse` are executed only once with running tests.
But adding filter function seems like attaching new hook with each executed test.

```
const filterFunction = function (request) {
    return !/^https?:\/\/\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/.test(request.url);
};

const hook = new RequestLogger(filterFunction);
fixture('Test').requestHooks(hook);

----

import { RequestHook } from 'testcafe';

export class RequestLogger extends RequestHook {
    constructor(requestFilterRuleInit) {
        super(requestFilterRuleInit, DEFAULT_OPTIONS);
    }

    onRequest(event) {
       // This get's called as many times as number executed tests in run
    }

    onResponse(event) {
       // This get's called as many times as number executed tests in run
  }
}
```

### What is the expected behavior?
I would expect that since it's the same `const` hook it won't get attached more than once.

### How would you reproduce the current behavior (if this is a bug)?
See below

#### Provide the test code and the tested page URL (if applicable)
https://github.com/andrzej-kodify/testcafe-requesthook-bug

### Specify your

* operating system: OSX
* testcafe version: 0.20.5
* node.js version: v10.5.0

### PR DESCRIPTION
**Pull request by miherlosev**
@AndreyBelym

### CODE REVIEW

File: src/api/request-hooks/hook.js [hunk 1]
```diff
@@ -17,12 +17,13 @@ export default class RequestHook {
         return [RequestFilterRule.ANY];
     }
 
-    _instantiateRequestFilterRules () {
+    instantiateRequestFilterRules () {
```
Comment (helen-dikareva) [inline src/api/request-hooks/hook.js]: let's leave it "private": `_instantiateRequestFilterRules`

Comment (helen-dikareva) [reply src/api/request-hooks/hook.js]: But the behavior didn't change.
Plus, I think better to leave private methods with private indicator `_` in public API class.

Comment (miherlosev) [reply src/api/request-hooks/hook.js]: Ok. Let it be.

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
Travis tasks are [green](https://travis-ci.org/testcafe-build-bot/testcafe/builds/426693454?utm_source=github_status&utm_medium=notification). But [comment](https://github.com/DevExpress/testcafe/pull/2834#issuecomment-420046964) is red. If same situation will repeat then it needs to investigate.

**Pull request merged**